src/csharp/Microsoft.Spark/Sql/RowCollector.cs

```diff
@@ -37,5 +38,81 @@ public IEnumerable<Row> Collect(ISocketWrapper socket)
                 }
             }
         }
+
+        /// <summary>
+        /// Synchronously collects pickled row objects from the given socket.
+        /// </summary>
+        /// <param name="socket">Socket the get the stream from</param>
+        /// <param name="server">The jvm socket auth server</param>
+        /// <returns>Collection of row objects</returns>
+        public IEnumerable<Row> SynchronousCollect(ISocketWrapper socket, JvmObjectReference server) =>
```

Comment:
This can be named as `Collect` as an overload? The existing `Collect` seems to be synchronous as well. And you can add more descriptive comments to summary that this API collects rows by partitions leveraging the other `Collect`, for example.